### PR TITLE
Renovation: get rid of ts warnings

### PR DESCRIPTION
**Pull request by churkin**

### CODE REVIEW

File: js/renovation/ui/button.tsx [hunk 1]
```diff
@@ -40,19 +41,18 @@ const getCssClasses = (model: ButtonProps) => {
...
 };
-
-export const viewFunction = (viewModel: Button) => {
+export const viewFunction = (viewModel: Button): JSX.Element => {
   const {
-    children, icon, iconPosition, template, text,
+    children, icon, iconPosition, template: ButtonTemplate, text,
   } = viewModel.props;
-  const renderText = !template && !children && text;
+  const renderText = !ButtonTemplate && !children && text;
```
Comment (vconst) [inline js/renovation/ui/button.tsx:48]: WTF? Why are tests passed?

File: js/renovation/ui/button.tsx [hunk 2]
```diff
@@ -40,19 +41,18 @@ const getCssClasses = (model: ButtonProps) => {
...
   const isIconLeft = iconPosition === 'left';
-  const iconComponent = !template && !children && viewModel.iconSource
+  const iconComponent = !ButtonTemplate && !children && viewModel.iconSource
```
Comment (vconst) [inline js/renovation/ui/button.tsx:50]: WTF?

File: js/renovation/ui/button.tsx [hunk 3]
```diff
@@ -74,20 +74,20 @@ export const viewFunction = (viewModel: Button) => {
...
     >
       <div className="dx-button-content" ref={viewModel.contentRef as any}>
-        {template
-                && (
-                <viewModel.props.template
-                  data={{ icon, text }}
-                />
-                )}
-        {!template && children}
+        {ButtonTemplate && (<ButtonTemplate data={{ icon, text }} />)}
```
Comment (vconst) [inline js/renovation/ui/button.tsx:77]: Why is template replaced to ButtonTemplate?

File: js/renovation/ui/button.tsx [hunk 4]
```diff
@@ -74,20 +74,20 @@ export const viewFunction = (viewModel: Button) => {
...
+        {!ButtonTemplate && children}
```
Comment (vconst) [inline js/renovation/ui/button.tsx:78]: Why is template replaced to ButtonTemplate?

File: js/core/devices.d.ts [hunk 5]
```diff
@@ -149,6 +149,14 @@ declare class DevicesObject {
...
      */
     real(): Device;
+    /**
+     * @docid DevicesObjectMethods.isSimulator
+     * @publicName isSimulator()
+     * @return Boolean
+     * @prevFileNamespace DevExpress.core
+     * @hidden
+     */
+    isSimulator(): boolean;
```
Comment (vconst) [inline js/core/devices.d.ts:152]: Why is doc comment added here?

File: js/renovation/ui/button.tsx (hunk 1 above)
Comment (roman-simionov) [reply js/renovation/ui/button.tsx:48]: it is just a variable name

https://github.com/DevExpress/DevExtreme/pull/13873/files#diff-6bde74af2450bbb641a700bbb770adf2R46

**Review by roman-simionov — DISMISSED** [review]

File: js/renovation/ui/button.tsx (hunk 1 above)
Comment (churkin) [reply js/renovation/ui/button.tsx:48]: What is wrong?

File: js/renovation/ui/button.tsx (hunk 2 above)
Comment (churkin) [reply js/renovation/ui/button.tsx:50]: WTF?

File: js/renovation/ui/button.tsx (hunk 4 above)
Comment (churkin) [reply js/renovation/ui/button.tsx:78]: JSX requires upper case name for the component

File: js/core/devices.d.ts (hunk 5 above)
Comment (churkin) [reply js/core/devices.d.ts:152]: Where should I add they?

File: js/renovation/ui/button.tsx (hunk 1 above)
Comment (vconst) [reply js/renovation/ui/button.tsx:48]: It is not a type. I think that ButtonTemplate is type of template.

**Review by vconst — DISMISSED** [review]

File: js/core/devices.d.ts (hunk 5 above)
Comment (vconst) [reply js/core/devices.d.ts:152]: Previously, the isSimulator method was internal. We want to publish it? Usually, we do not write doc comments for internal methods.

Comment (churkin) [reply js/core/devices.d.ts:152]: It's marked as a "hidden" (https://github.com/DevExpress/DevExtreme/pull/13873/files#diff-5f66e6a1045e4cd65e0e6007dd944fe2R43) @IlyaKhD is everything good here?

Comment (IlyaKhD) [reply js/core/devices.d.ts:152]: If the intention is to hide a member from wrappers and other generators, then doc-tags shouldn't be added at all.

**Review by roman-simionov — APPROVED** [review]

**Pull request merged**